### DISCUSSION TITLE
Snapshot flash light for ESP32 CAM

### DISCUSSION THREAD

**Discussion by ELK169** [arendst/Tasmota#22061] [Ideas]
When capturing single images without a running (MJPEG) stream (either a single picture as a response to an external event or periodically captured time lapse or very-low-fps videos (i.e. with BlueIris)), it seems helpful to me to have a function to activate the integrated white LED automatically (as a flashlight) while capturing the image. So you can get a much better (color!) picture quality while at the same time keeping the surrounding dark in the night. 
As an extension, the used pin could/should be definable. A dimming function using PWM might be problematic.

**Comment by sfromis** [comment]
If you do the normal config of the led pin as `PWM` you can use it the same way as with fixed level config as a `Relay`, with the addition of variable brightness; in both cases you can issue a command to turn it on shortly before the snapshot. Having `PulseTime` configured already allows you to set an interval after which it is switched off again.

**Reply by whitedavidp** [reply]
@sfromis, Can you please give an example of the command(s) needed to do this?

**Reply by sfromis** [reply]
If you config the led as a `Relay` pin, it will work as a lamp with the usual `Power` on/off commands, and `Dimmer` for brightness. And if you want it to automatically power off after an interval, the `PulseTime` command can be used to set that up.

However, this does not automatically work with snapshotting an image, meaning that you'd have to turn on the light *before* sampling the snapshot image.

**Reply by whitedavidp** [reply]
Thanks for the response. I plan to turn the light on, grab a snapshot via http, and turn it off.

I am now able to user power1 on/off and it works! Thanks so much.

Any chance you know how to turn off the solid, red LED on the camera MB?

![IMG_20250504_131054444](https://github.com/user-attachments/assets/35c8c161-9138-4aa1-8555-92d61d7feaa9)

Thanks!

**Reply by sfromis** [reply]
As that led is unrelated to the camera (no such connections to the USB/serial adapter "MB" board), darkening it is by physical means, covering/removing/destroying, or cutting a PCB trace.

You might also want to consider if you even want to have that adapter board in regular use, as you could power the camera board with 5V without that extra board.

**Comment by whitedavidp** [comment]
Right you are about avoiding it in use. I may do that depending on how I end up getting it into a case. I was more concerned about lessening power requirements than blocking the light source.  So I may end up going without it. Thanks so much for your help!
